include/ibamr/vc_ins_utilities.h

```diff
@@ -0,0 +1,414 @@
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////////////// INCLUDE GUARD ////////////////////////////////
+
+#ifndef included_IBAMR_vc_ins_utilities
+#define included_IBAMR_vc_ins_utilities
+
+#include <ibtk/config.h>
+
+#include <ibamr/AdvDiffHierarchyIntegrator.h>
+#include <ibamr/INSVCStaggeredHierarchyIntegrator.h>
+#include <ibamr/ibamr_enums.h>
+
+#include "ibtk/CartGridFunction.h"
+
+#include "tbox/Pointer.h"
+
+namespace IBTK
+{
+class HierarchyMathOps;
+}
+
+/////////////////////////////// FUNCTION DEFINITIONS /////////////////////////
+namespace IBAMR
+{
+/*!
+ * \brief The vc_ins_utilities class can be utilized to set fluid properties such as density and viscosity
+ * for both two-phase and three-phase flows throughout the entire domain. Additionally, this class provides
+ * implementations for gravity force calculations in two-phase and three-phase flows.
+ *
+ * \note Various options are available for computing the side-centered density within this class.
+ */
+namespace VCINSUtilities
+{
+
+/*!
+ * Pre processing call back function to be hooked into IBAMR::AdvDiffHierarchyIntegrator class.
+ *
+ * \param rho_idx a patch data index for the current density variable maintained by the integrator.
+ * \param ctx is the pointer to SetFluidProperties class object.
+ */
+
+void callSetDensityCallbackFunction(int rho_idx,
+                                    SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM> > rho_var,
+                                    SAMRAI::tbox::Pointer<IBTK::HierarchyMathOps> hier_math_ops,
+                                    const int cycle_num,
+                                    const double time,
+                                    const double current_time,
+                                    const double new_time,
+                                    void* ctx);
+
+/*!
+ * Pre processing call back function to be hooked into IBAMR::AdvDiffHierarchyIntegrator class.
+ *
+ * \param rho_idx a patch data index for the current density variable maintained by the integrator.
+ * \param ctx is the pointer to SetFluidProperties class object.
+ */
+
+void callSetViscosityCallbackFunction(int mu_idx,
+                                      SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM> > mu_var,
+                                      SAMRAI::tbox::Pointer<IBTK::HierarchyMathOps> hier_math_ops,
+                                      const int cycle_num,
+                                      const double time,
+                                      const double current_time,
+                                      const double new_time,
+                                      void* ctx);
+
+class SetFluidProperties
+{
+    /*!
+     * \brief Class SetFluidProperties is a utility class which sets the fluid and
+     * solid Eulerian density based on the current level set information
+     */
```

Comment:
Comment should go above class definition.
```suggestion
/*!
 * \brief Class SetFluidProperties is a utility class which sets the fluid and
 * solid Eulerian density based on the current level set information
 */
class SetFluidProperties
{
```